packages/flutter_tools/test/integration.shard/isolated/native_assets_without_cbuild_assemble_test.dart

```diff
@@ -0,0 +1,157 @@
+    platform.operatingSystem,
+
+    // On macOS, also test iOS.
+    if (platform.isMacOS) 'ios',
+
+    // On every host platform, test Android.
+    'apk',
+  ]) {
+    _testBuildCommand(
+      buildCommand: buildCommand,
+      processManager: processManager,
+      nativeAssetsCliVersionConstraint: constraint,
+    );
+  }
+}
+
+void _testBuildCommand({
+  required String buildCommand,
+  required String nativeAssetsCliVersionConstraint,
+  required ProcessManager processManager,
+}) {
+  testWithoutContext(
+    'flutter build "$buildCommand" succeeds without libraries',
+    () async {
+      await inTempDir((Directory tempDirectory) async {
+        const String packageName = 'uses_package_native_assets_cli';
+
+        // Create a new (plain Dart SDK) project.
+        await expectLater(
+          processManager.run(
+            <String>[
+              flutterBin,
+              'create',
+              '--no-pub',
+              packageName,
+            ],
+            workingDirectory: tempDirectory.path,
+          ),
+          completion(const ProcessResultMatcher()),
+        );
+
+        final Directory packageDirectory = tempDirectory.childDirectory(
+          packageName,
+        );
+
+        // Add native_assets_cli and resolve implicitly (pub add does pub get).
+        // See https://dart.dev/tools/pub/cmd/pub-add#version-constraint.
+        await expectLater(
+          processManager.run(
+            <String>[
+              flutterBin,
+              'packages',
+              'add',
+              'native_assets_cli:$nativeAssetsCliVersionConstraint',
+            ],
+            workingDirectory: packageDirectory.path,
+          ),
+          completion(const ProcessResultMatcher()),
+        );
+
+        // Add a build hook that does nothing to the package.
+        packageDirectory.childDirectory('hook').childFile('build.dart')
+          ..createSync(recursive: true)
+          ..writeAsStringSync('''
+import 'package:native_assets_cli/native_assets_cli.dart';
+
+void main(List<String> args) async {
+  await build(args, (config, output) async {});
+}
+''');
+
+        // Try building.
+        await expectLater(
+          processManager.run(
+            <String>[
+              flutterBin,
+              'build',
+              'macos',
+              '--debug',
+            ],
+            workingDirectory: packageDirectory.path,
+          ),
+          completion(const ProcessResultMatcher()),
+        );
+      });
+    },
+  );
+}
+
+/// Reads `templates/package_ffi/pubspec.yaml.tmpl` to use the package version.
+///
+/// For example, if the template would output:
+/// ```yaml
+/// dependencies:
+///   native_assets_cli: ^0.8.0
+/// ```
+///
+/// ... theh this function would return `'^0.8.0'`.
+String _getPackageFfiTemplatePubspecVersion() {
+  final String path = Context().join(
+    getFlutterRoot(),
+    'packages',
+    'flutter_tools',
+    'templates',
+    'package_ffi',
```

Comment:
Just to note: This test isn't FFI-related, but I like that we can share the test version pin.

Reply:
Agreed!